types/fs-base-ui/html_ui/JS/common.d.ts

```diff
@@ -0,0 +1,1102 @@
+        function absolutePath(current: string, relativePath: string): string;
+        function absoluteURL(current: string, relativePath: string): string;
+        function setAsyncLoading(async: boolean): void;
+        function onAllResourcesLoaded(): void;
+    }
+
+    namespace LiveReload {
+        function reloadCSS(): void;
+
+        /**
+         * Starts automatically reloading CSS when autoRefresh is true, stops the reloading when false.
+         */
+        function startAutoReload(autoRefresh: boolean): void;
+    }
+
+    // Coherent is assigned in fs-base-ui/html_ui/JS/common.js, however as all the functions are in
+    // fs-base-ui/html_ui/JS/coherent.js we list them there and not here.
+
+    let debugDuplicateRegister: boolean;
+    let debugLocalization: boolean;
+    function cancelAllRequestAnimationFrame(): void;
+
+    namespace CoherentSetup {
+        function CheckCoherentEngine(windowElem: Window): void;
+        function updateScreenSize(): void;
+    }
+
+    namespace Utils {
+        function dispatchToAllWindows(event: Event): void;
+
+        /**
+         * Converts an arrayLike object into an array.
+         */
+        function toArray<T>(arrayLike: ArrayLike<T>): T[];
+
+        /**
+         * Determines if the window is running in an iframe.
+         */
+        function inIframe(): boolean;
+
+        /**
+         * Creates a div element with the given classes.
+         */
+        function createDiv(...classList: string[]): HTMLDivElement;
+
+        /**
+         * Gets the given percentage of the virtual height, suffixed with "px".
+         */
+        function getVh(percent: number): string;
+
+        /**
+         * Gets the size in pixels adapted to the virtual height.
+         * Internally it seems the application works with 1080px.
+         */
+        function getSize(px: number): number;
+
+        /**
+         * Gets the given percentage of the virtual height.
+         */
+        function getVhNumber(percent: number): number;
+
+        function getScreenRatio(): number;
+
+        /**
+         * Gets the screen height of the window.
+         */
+        function getVirtualHeight(): number;
+
+        /**
+         * Determines if the scroll bar is visible for the given element.
+         */
+        function scrollbarVisible(element: Element): boolean;
+
+        /**
+         * Gets a CSS URL string, where any \ in the the url parameter is replaced with / and the URL is prefixed with
+         * the given prefix.
+         */
+        function getExternalImageUrl(url: string, prefix?: string): string;
+
+        /**
+         * Returns the remainder of one number divided by another.
+         */
+        function Modulo(num: number, mod: number): number;
+
+        /**
+         * Returns the number with as many "0" prefixes as needed to get to a string of the given length.
+         */
+        function pad(num: number, length: number): string;
```

Comment:
Would also for other types for `num` than number, e. g. if `num` was a string.

Reply:
I'll add `string` to the types. I don't think adding any other type would be in line with the "thought" behind this function.